src/analysis/processing/qgsalgorithmfindprojection.cpp

```diff
@@ -0,0 +1,180 @@
+{
+  return QObject::tr( "crs,srs,coordinate,reference,system,guess,estimate,finder,determine" ).split( ',' );
+}
+
+QString QgsFindProjectionAlgorithm::group() const
+{
+  return QObject::tr( "Vector general" );
+}
+
+QString QgsFindProjectionAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeneral" );
+}
+
+QString QgsFindProjectionAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Creates a list of possible candidate coordinate reference systems for a layer "
+                      "with an unknown projection.\n\n"
+                      " The expected area which the layer should reside in must be specified via the "
+                      "target area parameter.\n\n"
+                      "The algorithm operates by testing the layer's extent in every known reference "
+                      "system and listing any in which the bounds would fall near the target area if "
+                      "the layer was in this projection." );
+}
+
+QgsFindProjectionAlgorithm *QgsFindProjectionAlgorithm::createInstance() const
+{
+  return new QgsFindProjectionAlgorithm();
+}
+
+void QgsFindProjectionAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ) ) );
+  addParameter( new QgsProcessingParameterExtent( QStringLiteral( "TARGET_AREA" ), QObject::tr( "Target area for layer" ) ) );
+
+  // deprecated
+  auto crsParam = std::make_unique<QgsProcessingParameterCrs>( QStringLiteral( "TARGET_AREA_CRS" ), QObject::tr( "Target area CRS" ), QVariant(), true );
+  crsParam->setFlags( crsParam->flags() | Qgis::ProcessingParameterFlag::Hidden );
+  addParameter( crsParam.release() );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "CRS candidates" ), Qgis::ProcessingSourceType::Vector ) );
+}
+
+QVariantMap QgsFindProjectionAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  std::unique_ptr<QgsProcessingFeatureSource> source( parameterAsSource( parameters, QStringLiteral( "INPUT" ), context ) );
+  if ( !source )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  const QgsRectangle extent = parameterAsExtent( parameters, QStringLiteral( "TARGET_AREA" ), context );
+  QgsCoordinateReferenceSystem targetCrs = parameterAsExtentCrs( parameters, QStringLiteral( "TARGET_AREA" ), context );
+  if ( parameters.contains( QStringLiteral( "TARGET_AREA_CRS" ) ) )
+  {
+    QgsCoordinateReferenceSystem crs = parameterAsCrs( parameters, QStringLiteral( "TARGET_AREA_CRS" ), context );
+    if ( crs.isValid() )
+    {
+      targetCrs = crs;
+    }
+  }
+
+  QgsGeometry targetGeometry = QgsGeometry::fromRect( extent );
+
+  QgsFields fields;
+  fields.append( QgsField( "auth_id", QMetaType::Type::QString, "", 20 ) );
```

Comment:
Let's drop this -- there's potentially longer authids since this was originally written, and not much benefit from setting a fixed width anyway:

```suggestion
  fields.append( QgsField( "auth_id", QMetaType::Type::QString ) );
```